src/main/src/main/java/org/geoserver/catalog/impl/CatalogImpl.java

```diff
@@ -150,7 +146,10 @@ public void setFacade(CatalogFacade facade) {
         if (configurationLock != null) {
             facade = LockingCatalogFacade.create(facade, configurationLock);
         }
-        this.facade = facade;
+        // wrap the default catalog facade with the facade capable of handling isolated workspaces
+        // behavior
+        this.facade = new IsolatedCatalogFacade(facade);
+        this.facade.setCatalog(this);
         facade.setCatalog(this);
```

Comment:
I believe this has changed the nesting structure, which could have side effects with concurrent usage:

* Before: ```LockingCatalogFacade(IsolatedCatalogFacade(DefaultCatalogFacade)))```
* After:  ```IsolatedCatalogFacade(LockingCatalogFacade(DefaultCatalogFacade)))```

Reply:
Indeed, I fixed it.